gsw/GENERIC_TORQUER/lib/generic_torquer_lib.rb

```diff
@@ -0,0 +1,121 @@
+# Library for GENERIC_TORQUER Target
+require 'cosmos'
+require 'cosmos/script'
+
+#
+# Definitions
+#
+GENERIC_TORQUER_CMD_SLEEP = 0.25
+GENERIC_TORQUER_RESPONSE_TIMEOUT = 5
+GENERIC_TORQUER_TEST_LOOP_COUNT = 1
+GENERIC_TORQUER_DEVICE_LOOP_COUNT = 5
+
+#
+# Functions
+#
+def get_generic_torquer_hk()
+    cmd("GENERIC_TORQUER GENERIC_TORQUER_REQ_HK_CC")
+    wait_check_packet("GENERIC_TORQUER", "GENERIC_TORQUER_HK_TLM_T", 1, GENERIC_TORQUER_RESPONSE_TIMEOUT)
+    sleep(GENERIC_TORQUER_CMD_SLEEP)
+end
+
+def generic_torquer_cmd(*command)
+    count = tlm("GENERIC_TORQUER GENERIC_TORQUER_HK_TLM_T CMD_COUNT") + 1
+
+    if (count == 256)
+        count = 0
+    end
+
+    cmd(*command)
+    get_generic_torquer_hk()
+    current = tlm("GENERIC_TORQUER GENERIC_TORQUER_HK_TLM_T CMD_COUNT")
+    if (current != count)
+        # Try again
+        cmd(*command)
+        get_generic_torquer_hk()
+        current = tlm("GENERIC_TORQUER GENERIC_TORQUER_HK_TLM_T CMD_COUNT")
+        if (current != count)
+            # Third times the charm
+            cmd(*command)
+            get_generic_torquer_hk()
+            current = tlm("GENERIC_TORQUER GENERIC_TORQUER_HK_TLM_T CMD_COUNT")
+        end
+    end
+    check("GENERIC_TORQUER GENERIC_TORQUER_HK_TLM_T CMD_COUNT >= #{count}")
+end
+
+def enable_generic_torquer()
+    # Send command
+    generic_torquer_cmd("GENERIC_TORQUER GENERIC_TORQUER_ENABLE_CC")
+    # Confirm
+    check("GENERIC_TORQUER GENERIC_TORQUER_HK_TLM_T DEVICE_ENABLED == 'ENABLED'")
+end
+
+def disable_generic_torquer()
+    # Send command
+    generic_torquer_cmd("GENERIC_TORQUER GENERIC_TORQUER_DISABLE_CC")
+    # Confirm
+    check("GENERIC_TORQUER GENERIC_TORQUER_HK_TLM_T DEVICE_ENABLED == 'DISABLED'")
+end
+
+def safe_generic_torquer()
+    get_generic_torquer_hk()
+    state = tlm("GENERIC_TORQUER GENERIC_TORQUER_HK_TLM_T DEVICE_ENABLED")
+    if (state != "DISABLED")
+        disable_generic_torquer()
+    end
+end
+
+def confirm_generic_torquer_data()

```

Comment:
The tests performed here are really all we can do at the TRQ level. Pushing these up to ADCS seems fine to me.